ResidentContactApi.Tests/V1/Controllers/ResidentContactApiControllerTest.cs

```diff
@@ -180,5 +184,49 @@ public void InsertResidentReturns500StatusCodeIfExternalReferenceNotInsertExcept
             result.StatusCode.Should().Be(500);
             result.Value.Should().Be("External reference could not be inserted - error message");
         }
+
+        [Test]
+        public void IfExternalResidentToBeInsertedAlreadyExistsShouldReturn200StatusCode()
+        {
+            var response = new InsertResidentResponse { ResidentRecordAlreadyPresent = true };
+            _mockInsertExternalReferenceRecordUseCase.Setup(x => x.Execute(It.IsAny<InsertResidentRequest>())).Returns(response);
+
+            var result = _classUnderTest.InsertExternalResident(It.IsAny<InsertResidentRequest>()) as OkObjectResult;
```

Comment:
Name is a bit confusing. Is this inserting an external reference to a record or a new resident record?